pandas/core/reshape/pivot.py

```diff
@@ -422,13 +423,18 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Collection[Label]]] = None,
+    columns: Union[Label, List[Label]] = None,
```

Comment:
Should this also be Optional?

Reply:
yeah, i am not very sure, got some mypy errors below to complain that `columns` is not Optional, probably because we will raise error if `None`.